src/OpenTelemetry.Exporter.OpenTelemetryProtocol/TraceExporter.cs

```diff
@@ -14,188 +14,67 @@
 // limitations under the License.
 // </copyright>
 
-namespace OpenTelemetry.Exporter.OpenTelemetryProtocol
-{
-    using System;
-    using System.Collections.Concurrent;
-    using System.Collections.Generic;
-    using System.Diagnostics;
-    using System.Reflection;
-    using System.Threading;
-    using System.Threading.Tasks;
-
-    using Google.Protobuf.WellKnownTypes;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 
-    using Grpc.Core;
+using Grpc.Core;
 
-    using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
-    using OpenTelemetry.Trace;
-    using OpenTelemetry.Trace.Export;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
+using OpenTelemetry.Trace.Export;
 
-    using Proto = Opentelemetry.Proto;
+using OtlpCollector = Opentelemetry.Proto.Collector.Trace.V1;
+using OtlpCommon = Opentelemetry.Proto.Common.V1;
+using OtlpResource = Opentelemetry.Proto.Resource.V1;
+using OtlpTrace = Opentelemetry.Proto.Trace.V1;
 
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol
+{
     /// <summary>
-    /// The trace exporter to otelcol.
+    /// The trace exporter using the OpenTelemetry protocol (OTLP).
     /// </summary>
-    // TODO support async exporting and results
-    public class TraceExporter : SpanExporter, IDisposable
+    public class TraceExporter : SpanExporter
     {
-        private const uint MaxSpanBatchSize = 32;
         private readonly Channel channel;
-        private readonly Proto.Agent.Trace.V1.TraceService.TraceServiceClient traceClient;
-        private readonly ConcurrentQueue<SpanData> spans = new ConcurrentQueue<SpanData>();
-        private readonly Proto.Agent.Common.V1.Node node;
-        private readonly uint spanBatchSize;
-        private CancellationTokenSource cts;
-        private Task runTask;
+        private readonly OtlpCollector.TraceService.TraceServiceClient traceClient;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TraceExporter"/> class.
         /// </summary>
-        /// <param name="agentEndpoint">The agent endpoint.</param>
-        /// <param name="hostName">Name of the host.</param>
-        /// <param name="serviceName">Name of the service.</param>
-        /// <param name="credentials">The credentials.</param>
-        /// <param name="spanBatchSize">Size of the span batch.</param>
-        public TraceExporter(string agentEndpoint, string hostName, string serviceName, ChannelCredentials credentials = null, uint spanBatchSize = MaxSpanBatchSize)
```

Comment:
I wonder when you removed spanBatchSize?

Reply:
I removed the internal queue of the export and plan to recommend it to be used together with the batching processor - this pattern is also used on some other exporters.